### PR TITLE
Tensorboard callbacks

### LINKED ISSUE #123: Tensorboard callback
Implement tensorboard callback for logging metrics

### PR DESCRIPTION
**Pull request by arvindmuralie77**
#### Changes

Fixes #123 

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Did you update CHANGELOG (docs/CHANGELOG.md) in case of a major change?

Solves issue #123 
I have added basic tensorboard callback functionality which include: logging the loss and accuracy for both training and validation.
Is there any other functionality of tensorboard that I need to add @aniketmaurya

### CODE REVIEW

File: gradsflow/callbacks/tensorboard.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            comment=self.comment,
+            purge_step=self.purge_step,
+            flush_secs=self.flush_secs,
+            filename_suffix=self.filename_suffix,
+        )
+
+    def on_train_epoch_end(self):
+        tracker = self.model.tracker
+        self.writer.add_scalar("train/loss", scalar_value=tracker.train.loss.avg, global_step=tracker.current_epoch)
+        for metric, value in tracker.train.metrics.items():
```
Comment (aniketmaurya) [inline gradsflow/callbacks/tensorboard.py:49]: `tracker.train_metrics.to_dict()` will ensure all the values are converted from torch.Tensor to int/float type.

**Comment by aniketmaurya** [comment]
awesome work, feel free to set the PR for review when you're ready.

**Comment by aniketmaurya** [comment]
> @aniketmaurya set the PR as ready for review

Looks good! Do wanna add logging for step_loss as well in this PR or maybe a next PR?

**Comment by arvindmuralie77** [comment]
Sure I'll add step_loss in this PR

**Comment by arvindmuralie77** [comment]
Hey, @aniketmaurya should I also plot the metrics for every step epoch?

**Comment by aniketmaurya** [comment]
> Hey, @aniketmaurya should I also plot the metrics for every step epoch?

Sure, let's log metrics for every `train_step`

**Comment by aniketmaurya** [comment]
You can refer to the Wandb callback for reference.

**Review by aniketmaurya — APPROVED** [review]
LGTM 👍

**Comment by arvindmuralie77** [comment]
@aniketmaurya added functionality loggging metrics and loss for every `step_loss` as well

**Review by aniketmaurya — APPROVED** [review]
Awesome work! Let's merge this now.

**Pull request merged**